### PR TITLE
add documentation for new app builder overview dashboard

### PR DESCRIPTION
**Pull request by ameesha**
### What does this PR do? What is the motivation?
A description of the automatic dashboard that will be available to App Builder customers. 
Project details here
https://datadoghq.atlassian.net/wiki/spaces/APPS/pages/4117201157/App+Builder+Out-of-the-box+OOTB+Dashboard

Staging dashboard here
https://dd.datad0g.com/dash/integration/32597/app-builder-overview?fromUser=false&refresh_mode=sliding&from_ts=1729711778191&to_ts=1729798178191&live=true

### Merge instructions
Should only be be merged once the feature is released and available (aiming for late next week).
- [ ] Please merge after reviewing

### Additional notes
This is based off the Workflows documentation for their dashboard
https://docs.datadoghq.com/service_management/workflows/#workflows-overview-dashboard

### CODE REVIEW

File: content/en/service_management/app_builder/_index.md [hunk 1]
```diff
@@ -59,6 +59,12 @@ To use the app with your data, click **+ Connect Data**, then either create a ne
 
 You can change the selected connection by clicking **Change Connection** in the app.
 
+## App Builder Overview dashboard
```
Comment (ameesha) [inline content/en/service_management/app_builder/_index.md:62]: the dashboard is called "App Builder Overview" so i think this capitalization is ok, but calling out cause linting disagrees

Comment (drichards-87) [reply content/en/service_management/app_builder/_index.md:62]: Yes, I agree, as this is the title of the dashboard.

File: content/en/service_management/app_builder/app-builder-dashboard.png
Comment (drichards-87) [inline content/en/service_management/app_builder/app-builder-dashboard.png:1]: Images are referenced from the /static/images folder, so you'll need to move the image there, possibly in /static/images/service_management/app_builder/.

**Review by drichards-87 — CHANGES_REQUESTED** [review]
Left some feedback about moving the image to the static/images folder.

**Review by drichards-87 — APPROVED** [review]

**Pull request merged**